AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/StorageEngine.swift

```diff
@@ -208,31 +208,41 @@ final class StorageEngine: StorageEngineBehavior {
             completion(.failure(causedBy: dataStoreError))
         }
 
-        let wrappedCompletion: DataStoreCallback<M> = { result in
-            guard modelSchema.isSyncable, let syncEngine = self.syncEngine else {
-                completion(result)
-                return
-            }
-
-            guard case .success(let savedModel) = result else {
-                completion(result)
-                return
+        do {
+            try storageAdapter.transaction {
+                let result = self.storageAdapter.save(model, 
+                                                      modelSchema: modelSchema,
+                                                      condition: condition,
+                                                      eagerLoad: eagerLoad)
+                guard modelSchema.isSyncable else {
+                    completion(result)
+                    return
+                }
+
+                guard case .success(let savedModel) = result else {
+                    completion(result)
+                    return
+                }
+
+                guard let syncEngine else {
+                    let message = "No SyncEngine available to sync mutation event, rollback save."
+                    self.log.verbose("\(#function) \(message) : \(savedModel)")
+                    throw DataStoreError.internalOperation(
+                        message,
+                        "Save was interrupted. `DataStore.stop()` may have been called.",
```

Comment:
nit: update messaging to `DataStore.save() was interrupted. DataStore.stop() may have been called.`